src/ModelManager.cpp

```diff
@@ -444,36 +442,49 @@ void ModelManager::fetchRemoteModels() {
     if (isFetchingRemoteModels())
         return;
 
-    isFetchingRemoteModels_ = true;
-    emit fetchingRemoteModels();
-
-    QUrl url(kModelListUrl);
-    QNetworkRequest request(url);
-    QNetworkReply *reply = network_->get(request);
-    connect(reply, &QNetworkReply::finished, this, [=] {
-        switch (reply->error()) {
-            case QNetworkReply::NoError:
-                parseRemoteModels(QJsonDocument::fromJson(reply->readAll()).object());
-                break;
-            default:
-                emit error(reply->errorString());
-                break;
-        }
+    QStringList repos = repositories_.getRepos();
+    for (auto&& urlString : repos) {
+        isFetchingRemoteModels_ = true;
+        emit fetchingRemoteModels();
+
+        QUrl url(urlString);
+        QNetworkRequest request(url);
+        QNetworkReply *reply = network_->get(request);
+        connect(reply, &QNetworkReply::finished, this, [=] {
+            switch (reply->error()) {
+                case QNetworkReply::NoError:
+                    parseRemoteModels(QJsonDocument::fromJson(reply->readAll()).object());
+                    break;
+                default:
+                    QString errstr = QString("Error fetching remote repository: ") + urlString +
+                            QString("\nError code: ") + reply->errorString() +
+                            QString("\nPlease double check that the address is reachable.");
+                    emit error(errstr);
+                    break;
+            }
 
-        isFetchingRemoteModels_ = false;
-        emit fetchedRemoteModels();
+            isFetchingRemoteModels_ = false;
+            emit fetchedRemoteModels();
 
-        reply->deleteLater();
-    });
+            reply->deleteLater();
+        });
+    }
 }
 
 void ModelManager::parseRemoteModels(QJsonObject obj) {
     using namespace translateLocally::models;
-    remoteModels_.clear();
     
+    bool empty = true;
     for (auto&& arrobj : obj["models"].toArray()) {
+        empty = false;
         QJsonObject obj = arrobj.toObject();
-        remoteModels_.append(parseModelInfo(obj, Remote));
+        Model remoteModel = parseModelInfo(obj, Remote);
+        if (!remoteModels_.contains(remoteModel)) { // This costs O(n). Not happy, is there a better way?
```

Comment:
The previous approach was to first clear `remoteModels_` and then add them again. That being sad I didn't think of having per repo remote models yet. Maybe models shouldn't live together unless they belong to the same repo, but that complicates the model listing logic (Though doable, since I'm refactoring that next anyways...) Would that be preferable?